Added extended auto-QA features (#998)
webrecorder/browsertrix-crawler: docs/docs/user-guide/qa.md

- When comparing `urn:view:<url>` records for crawl and replay, the screenshots are 95% similar.
- When comparing `urn:text:<url>` records from crawl and replay WACZs, the text is 90% similar.
- When comparing `urn:pageinfo:<url>` resource entries from crawl and replay, the crawl record had 10 good responses (2xx/3xx status code) and 0 bad responses (4xx/5xx status code), while replay had 9 good and 1 bad.

## QA Policies and Additional Options

A few additional options and policies exist in order to fine-tune the QA process. These are:

  1. The ability to set a maximum page count for the QA,

>   2. The ability to choose between 3 different QA algorithms.

Reviewer: ```suggestion
- The ability to set a maximum page count for the QA run.
- The ability to choose between 3 different QA algorithms.
```

Minor copyediting

  
These algorithms are: *linear* (first come first serve), *regex* (define a regular expression to perform QA only on URLs that match it), and *random* (define a probability for each page to be QA'd).

You can use the following CLI arguments for this:

  - `qaMaxUrls`: the maximum number of pages to perform QA on,
  - `qaPolicy`: can be one of `linear`, `regex` or `random`,
  - `qaRegex`: if `qaPolicy` is `regex`, then you can define your regular expression here,

>   - `qaProbability`: if `qaPolicy` is `random`, you can define your per-page QA probability here. This is a floating-point number between 0 and 1.

Reviewer: ```suggestion
  - `qaMaxUrls`: the maximum number of pages to perform QA on.
  - `qaPolicy`: can be one of `linear`, `regex` or `random`.
  - `qaRegex`: if `qaPolicy` is `regex`, then you can define your regular expression here.
  - `qaProbability`: if `qaPolicy` is `random`, you can define your per-page QA probability here. This is a floating-point number between 0 and 1.
```

  
### QA Policy: `linear`

In this QA mode, the first `qaMaxUrls` pages in the `pages.jsonl` file(s) will be scanned. Example:

>     --qaPolicy "linear" --qaMaxUrls 50"

Reviewer: ```suggestion
    --qaPolicy "linear" --qaMaxUrls 50
```

    
### QA Policy: `regex`

In this QA mode, only the pages that match the regular expression in `qaRegex` will be scanned. Example:

>     --qaPolicy "regex" --qaRegex='^https:\/\/en\.wikipedia\.org\/wiki\/R.*$' --qaMaxUrls 50"

Reviewer: ```suggestion
    --qaPolicy "regex" --qaRegex='^https:\/\/en\.wikipedia\.org\/wiki\/R.*$' --qaMaxUrls 50
```

    
This will match all english Wikipedia articles that start with `R`.

### QA policy: `random`

In this QA mode, pages will be scanned with a probability equal to `qaProbability`. This is a floating-point number between `0` and `1`. Example:

>     --qaPolicy "random" --qaProbability 0.3 --qaMaxUrls 50"

Reviewer: ```suggestion
    --qaPolicy "random" --qaProbability 0.3 --qaMaxUrls 50
```

    
This policy allows to get a good overall impression of a harvest by scanning a random sample of it.
    
### Maximum number of pages to scan

In every case mentioned above, the number of pages that will be queued for scanning will be at most `qaMaxUrls`.

## Usage with the Browsertrix UI

> You can easily use these features with Browsertrix UI (formerly *Browsertrix Cloud*) in the following manner:

Reviewer: ```suggestion
A future release of [Browsertrix](https://github.com/webrecorder/browsertrix) will likely integrate these settings into the Browsertrix application's user interface. In the meantime, it is possible to use these features with a [self-hosted deployment](https://docs.browsertrix.com/deploy/) of Browsertrix in the following manner:
```

Provides a link to the self-hosting documentation and a bit more context on the fact that we do plan to support this in the UI at some point in the future.

  1. Compile the crawler as usual,
  2. Tag it as you wish and push it to local registry,
  3. Adapt the `crawler_channels` in your deployment's `local-config.yaml` file so that it points to the crawler in your registry,
  4. Add your QA policy and other parameters to the `crawler_extra_args` variable,
  5. (optional) Clear your Kubernetes/microk8s cache with `microk8s ctr images rm localhost:32000/<your-crawler-image>`

>   6. Reload your deployment.

Reviewer: ```suggestion
  1. Compile the crawler as usual.
  2. Tag it as you wish and push it to local registry.
  3. Adapt the `crawler_channels` in your deployment's `local-config.yaml` file so that it points to the crawler in your registry.
  4. Add your QA policy and other parameters to the `crawler_extra_args` variable.
  5. (optional) Clear your Kubernetes/microk8s cache with `microk8s ctr images rm localhost:32000/<your-crawler-image>`.
  6. Reload your deployment.
```
